datafusion/src/physical_plan/functions.rs

```diff
@@ -1428,6 +1460,423 @@ where
     })
 }
 
+pub(crate) fn create_immutable_impl(
```

Comment:
This is needed to make (most/all) builtin immutable?

Reply:
looks like lots of duplication between this and `create_physical_fun`, would be good to refactor and reuse the code so they don't get out of sync.